common/dynamicconfig/client.go

```diff
@@ -98,6 +98,7 @@ type (
 		TaskQueueType enumspb.TaskQueueType
 		ShardID       int32
 		TaskType      enumsspb.TaskType
+		Destination   string
```

Comment:
You're going to need changes to file_based_client.go to set this field in Constraints (and logging, etc.). You should add a test that goes all the way from the test yaml file to Get (don't need every combination, just one with the new constraint is fine)